### ISSUE TITLE
Add `secp256r1` curve host fns

### ISSUE THREAD

**Issue by dmkozh** [stellar/rs-soroban-env#807]
`secp256r1` seems to be a reasonably popular curve, so it would be nice to add host-side support for it.

**Comment by jayz22** [comment]
Related to https://github.com/stellar/rs-soroban-env/issues/684 so we may add them together.

**Comment by anupsdf** [comment]
Moved to Post Soroban V1.

**Comment by tomerweller** [comment]
secp256r1 could enable [passkey](https://fidoalliance.org/passkeys/) wallets without requiring [signature transmutation](https://medium.com/goldfinch-fi/next-gen-crypto-ux-how-we-made-defi-for-the-world-f8994c01adfa)

**Comment by tomerweller** [comment]
[EIP-7212](https://eips.ethereum.org/EIPS/eip-7212) adds motivation. Including these examples:
> Apple’s Secure Enclave: There is a separate “Trusted Execution Environment” in Apple hardware which can sign arbitrary messages and can only be accessed by biometric identification.

> Webauthn: Web Authentication (WebAuthn) is a web standard published by the World Wide Web Consortium (W3C). WebAuthn aims to standardize an interface for authenticating users to web-based applications and services using public-key cryptography. It is being used by almost all of the modern web browsers.

> Android Keystore: Android Keystore is an API that manages the private keys and signing methods. The private keys are not processed while using Keystore as the applications’ signing method. Also, it can be done in the “Trusted Execution Environment” in the microchip.

> Passkeys: Passkeys is utilizing FIDO Alliance and W3C standards. It replaces passwords with cryptographic key-pairs which is also can be used for the elliptic curve cryptography.

**Comment by leighmcculloch** [comment]
> `secp256r1`

For the benefit of anyone else who didn't immediately recognize this curve, in most APIs and documentation it is referred to as P256, or P-256, or NIST P-256, or Ecdsa P-256, and to a lesser frequency prime256v1.

It's one of the curves we discussed adding support to Stellar signers in [CAP-43](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0043.md). The same benefits and motivations detailed in that proposal would also apply here. The big one being support by hardware security modules.

**Comment by leighmcculloch** [comment]
> secp256r1 could enable [passkey](https://fidoalliance.org/passkeys/) wallets...

I think supporting just `secp256r1` would be sufficient to get decent passkey support.

I found some reports of some Windows machines only supporting RSA, and technically the Webauthn spec encourages at least supporting the following three schemes/algos:

>- -8 (Ed25519)
>- -7 (ES256)
>- -257 (RS256)

Ref: https://w3c.github.io/webauthn/#dom-publickeycredentialcreationoptions-pubkeycredparams

Where:
- -7 (ES256) is `secp256r1`
- -257 (RS256) is RSASSA-PKCS1-v1_5 with SHA-256 [RFC8017].

[RFC8017]: https://datatracker.ietf.org/doc/html/rfc8017

However, from best I can tell Ed25519 is not commonly supported, and a test we did confirmed ES256 is available on Windows in browners.

Also, supporting RSA would be challenging from a cost perspective as the keys and signatures are large. Keys and signatures would be 256-512 bytes.

We could support a limited set of passkeys that already support Ed25519 today, such as some Yubikeys.
